File: src/api-sdks/rcs-api/clients/csharp/src/IdeaStatiCa.RcsApi/RcsApiServiceAttacher.cs
using IdeaStatiCa.Api.Common;
using System.Threading.Tasks;

namespace IdeaStatiCa.RcsApi
{
	/// <summary>
	/// Factory for creating instances of RCS API client that are connected to the specified REST API service
	/// </summary>
	public class RcsApiServiceAttacher : IApiServiceFactory<IRcsApiClient>
	{
		string BaseUrl { get; set; }

		/// <summary>
		/// Constructor
		/// </summary>
		/// <param name="baseUrl"> URL of the REST API service</param>
		public RcsApiServiceAttacher(string baseUrl)
		{
			this.BaseUrl = baseUrl;
		}

		/// <inheritdoc cref="IApiServiceFactory{TClient}.CreateApiClient"/>
		public async Task<IRcsApiClient> CreateApiClient()
		{
			var client = new RcsApiClient(BaseUrl);
			await client.CreateAsync();
			return client;
		}
	}		
}

File: src/api-sdks/rcs-api/clients/python/ideastatica_rcs_api/rcs_api_service_attacher.py
import logging
from ideastatica_rcs_api.rcs_api_client import RcsApiClient

logger = logging.getLogger(__name__)

class RcsApiServiceAttacher:
    def __init__(self, base_url: str):
        self.base_url = base_url

    def create_api_client(self) -> RcsApiClient:
        logger.info(f"Creating client attached to {self.base_url}")
        return RcsApiClient(self.base_url) 